node/sc/bridge_test.go

```diff
@@ -665,3 +665,310 @@ func TestExtendedBridgeAndCallbackERC721(t *testing.T) {
+
+	tx, err = b.LockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+}
+
+// TestBridgeContract_TokenUnlockFail checks the following:
+// - testing the case unlocking token is fail.
+func TestBridgeContract_TokenUnlockFail(t *testing.T) {
+	backend, operator, _, b, _, _, erc20Addr, erc721Addr := generateBridgeTokenTestEnv(t)
+
+	// fail locking unregistered token.
+	testAddr := common.BytesToAddress([]byte("unregistered"))
+	tx, err := b.UnlockToken(operator, testAddr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+
+	// fail locking token if it is already unlocked.
+	tx, err = b.UnlockToken(operator, erc20Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+
+	tx, err = b.UnlockToken(operator, erc721Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+
+	tx, err = b.UnlockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusErrExecutionReverted, t)
+}
+
+// TestBridgeContract_CheckValueTransferAfterUnLock checks the following:
+// - the token can be unlock to allow value transfer requests.
+func TestBridgeContract_CheckValueTransferAfterUnLock(t *testing.T) {
+	//backend, operator, tester, b, erc20, erc721, erc20Addr, erc721Addr := generateBridgeTokenTestEnv(t)
+	backend, operator, tester, b, erc20, erc721, erc20Addr, erc721Addr := generateBridgeTokenTestEnv(t)
+
+	// lock token
+	tx, err := b.LockToken(operator, erc20Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.LockToken(operator, erc721Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.LockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	// unlock token
+	tx, err = b.UnlockToken(operator, erc20Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.UnlockToken(operator, erc721Addr)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	tx, err = b.UnlockKLAY(operator)
+	assert.NoError(t, err)
+	backend.Commit()
+	CheckReceipt(backend, tx, 1*time.Second, types.ReceiptStatusSuccessful, t)
+
+	// check value after unlocking
+	isLocked, err := b.LockedTokens(nil, erc20Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+
+	isLocked, err = b.LockedTokens(nil, erc721Addr)
+	assert.NoError(t, err)
+	assert.Equal(t, false, isLocked)
+
```

Comment:
Checking for KLAY is missing.
"""
	isLocked, err = b.IsLockedKLAY(nil)
	assert.NoError(t, err)
	assert.Equal(t, false, isLocked)
"""